ext/random/engine_mt19937.c

```diff
@@ -330,6 +330,7 @@ PHP_METHOD(Random_Engine_Mt19937, __serialize)
 	/* state */
 	array_init(&t);
 	if (!engine->algo->serialize(engine->status, Z_ARRVAL(t))) {
+		zval_ptr_dtor_nogc(&t);
 		zend_throw_exception(NULL, "Engine serialize failed", 0);
 		RETURN_THROWS();
 	}
```

Comment:
The entire error case is unreachable with the bundled engines. With non-bundled engines, I would also hope that serialization is infallible, but I think it would *theoretically* be possible for `t` to contain cycles. Perhaps just register `t` early in `return_value`?

```suggestion
	array_init(&t);
	zend_hash_next_index_insert(Z_ARRVAL_P(return_value), &t);
	if (!engine->algo->serialize(engine->status, Z_ARRVAL(t))) {
		zend_throw_exception(NULL, "Engine serialize failed", 0);
		RETURN_THROWS();
	}
```